editor/src/components/canvas/controls/select-mode-control-container.tsx

```diff
@@ -96,13 +91,14 @@ export class SelectModeControlContainer extends React.Component<
   static getDerivedStateFromProps(
     props: SelectModeControlContainerProps,
     previousState: SelectModeControlContainerState,
-  ) {
+  ): SelectModeControlContainerState {
```

Comment:
Unless your explicit intention is resetting lastHovered to null for every props change, I think this here should be `Partial<SelectModeControlContainerState>` and the `lastHovered: null` can be removed then.

Reply:
Ah, that would make more sense, I thought this might just be breezing through because `getDerivedStateFromProps` is effectively not checked that the types line up.

Reply:
Fixed in 3455633c.